tokio/src/runtime/blocking/pool.rs

```diff
@@ -177,7 +224,7 @@ impl Spawner {
 
             if shared.shutdown {
                 // Shutdown the task
-                task.shutdown();
+                task.task.shutdown();
```

Comment:
Perhaps include a comment here that it is ok to shut down a mandatory task here without running it because this only happens if the runtime has already started shutdown.

Reply:
Now that I think about it, we could still have a race where one worker thread successfully executes:

```
file.write(b"hello").await?;
```

but the runtime shutdown is scheduled from another thread, and the payload is never written to the file. The return value of `blocking::Spawner::spawn` (which conveys whether the task was actually scheduled) gets ignored in https://github.com/tokio-rs/tokio/pull/4316/files#diff-f1d42d2af51248e5dacb1cc3dd6bba12a956abba0d25567ee67bfd68d27e27d5R266

Our test doesn't show this error scenario because the `write` call and the runtime shutdown happen in the same thread, one after the other.

Perhaps `spawn_mandatory_blocking` should return an error when it tried to spawn a mandatory task but the runtime was already shutting down. That way, a call to `write` would only succeed if the write were to be actually attempted at shutdown. What do you think?

Reply:
FWIW, the current version of this PR (+ the comment you suggested) would already fix the original issue (#4296), so I leave it up to you whether you want to pursue the fix to https://github.com/tokio-rs/tokio/pull/4316#discussion_r781519904 in this PR or in a follow-up PR

Reply:
If it's simple to fix here, then go ahead and do it.

Reply:
I've taken a stab at fixing the problem.